gobblin-modules/gobblin-http/src/main/java/gobblin/utils/HttpUtils.java

```diff
@@ -0,0 +1,97 @@
+package gobblin.utils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.commons.lang3.text.StrSubstitutor;
+import org.apache.http.client.utils.URIBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+import gobblin.http.HttpOperation;
+
+
+/**
+ * Utilities to build gobblin http components
+ */
+public class HttpUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(HttpUtils.class);
+  private static final Gson GSON = new Gson();
+
+  /**
+   * Convert the given {@link GenericRecord} to {@link HttpOperation}
+   */
+  public static HttpOperation toHttpOperation(GenericRecord record) {
+    if (record instanceof HttpOperation) {
+      return (HttpOperation) record;
+    }
+
+    HttpOperation httpOperation = new HttpOperation();
+    boolean hasAnyField = false;
+    for (Schema.Field field: HttpOperation.getClassSchema().getFields()) {
```

Comment:
Note this will not work for nested schemas. That's ok, but it might be important to note that in the schema so future changes don't break the logic.